packages/js/plugins/uri-resolvers/ens-resolver/src/index.ts

```diff
@@ -125,22 +125,22 @@ export class EnsResolverPlugin extends Module<EnsResolverPluginConfig> {
         client
       );
 
-      if (error) {
-        throw error;
+      if (!result.ok) {
+        throw result.error;
       }
 
-      if (data) {
-        if (typeof data !== "string") {
+      if (result.value) {
+        if (typeof result.value !== "string") {
           throw Error(
-            `Malformed data returned from Ethereum.callContractView: ${data}`
+            `Malformed data returned from Ethereum.callContractView: ${result.value}`
           );
         }
 
-        return data;
+        return result.value;
       }
 
       throw Error(
-        `Ethereum.callContractView returned nothing.\nData: ${data}\nError: ${error}`
+        `Ethereum.callContractView returned nothing.\nData: \nError: `
```

Comment:
Should we have the `\nData: \nError:`? Might be miss leading but not sure

Reply:
I went back and forth on this. Ultimately I decided not to change the behavior.